### ISSUE TITLE
Import keeps pulling in the web version when used in React Native

### ISSUE THREAD

**Issue by kaxline** [react-native-async-storage/async-storage#631]
## You want to:  
Use the AsyncStorage exported from `/src/AsyncStorage.native.js`

## Details:
Maybe I just don't understand how webpack or metro works, but I keep getting the error 

`TypeError: undefined is not an object (evaluating 'window.localStorage.getItem')`

Which is thrown because the import is pulling in `/src/AsyncStorage.js`

I can't seem to figure out how this module would ever pull in `/src/AsyncStorage.native.js`. The `index.js` only ever pulls in the web version.

I'm sure I'm not getting something obvious, but I would appreciate any explanation that can be given.

**Comment by Krizzu** [comment]
Hey,

Is it pure React Native project? `metro` will respect the extension and import proper file per platform, unless you have a custom setup.

**Comment by kaxline** [comment]
@Krizzu No, it's an npm module that is supposed to detect what type of project it's been pulled into and use the right storage. So I want it to detect if it's in a React Native env and use this library if it is. Is that possible?

**Comment by tido64** [comment]
@Krizzu is right. If you're using Metro, it should've picked `AsyncStorage.native.js` for native platforms and `AsyncStorage.js` for web. I suspect something's wrong with your setup. Are you using Expo? Can you provide a minimal repro?

**Comment by kaxline** [comment]
@tido64 Sure, I'll work on getting a repro project. But would you be able to explain what triggers the use of the `AsyncStorage.native.js` file? Just for my own education. I can't figure it out.

**Comment by tido64** [comment]
When you execute `react-native bundle --platform ios`, the platform flag is passed to Metro so it knows what files to import when you call `import` or `require`. For `ios`, Metro will first check if there's a `.ios.js`, then `.native.js`, and finally `.js`, in that particular order. It will always pick the first file that exists. All your files are picked during bundle time. This does not happen at run time, as that would mean that all code for _all_ platforms are bundled with your app. That's just a waste of disk space. This is also why the issue you're seeing is specific to your bundle command or set up, and is unlikely to have anything to do with AsyncStorage.
